internal/providercache/package_install.go

```diff
@@ -37,12 +69,17 @@ func installFromHTTPURL(ctx context.Context, meta getproviders.PackageMeta, targ
 	// through X-Terraform-Get header, attempting partial fetches for
 	// files that already exist, etc.)
 
-	httpClient := httpclient.New()
+	retryableClient := retryablehttp.NewClient()
+	retryableClient.HTTPClient = httpclient.New()
+	retryableClient.RetryMax = 2
```

Comment:
You mentioned in the issue that this is hardcoded. And I think we should attempt to make this configurable similar to the retry discovery approach.

This is currently handled in the registry_client.go init() function here: https://github.com/opentofu/opentofu/blob/main/internal/getproviders/registry_client.go#L59

And that reads from an environment variable.

I think something like this should be added here too. Maybe with a default of 2 retries but configurable through an env var named something like `TF_PROVIDER_DOWNLOAD_RETRY`?

Reply:
does `init` method works with provider installer package ?

Reply:
It should work yes. You can find docs on the `init` method here: https://go.dev/doc/effective_go#init

Reply:
Oh, that's cool. I wasn't aware of that.